lib/search-field/index.jsx

```diff
@@ -36,21 +42,37 @@ export class SearchField extends Component {
     }
   }
 
-  interceptEsc = event => {
-    if (KEY_ESC === event.keyCode) {
-      if (this.state.query === '') {
-        this.inputField.blur();
-      }
-      this.clearQuery();
+  doSearch = query => {
+    this.setState({ query, searchSelected: true });
+    this.debouncedSearch(query);
+  };
+
+  interceptKeys = event => {
+    switch (event.keyCode) {
+      case KEY_ESC:
+        if (this.state.query === '') {
+          this.inputField.blur();
+        }
```

Comment:
some people don't like my style here but I don't like the way nesting slants the code

```js
case KEY_ESC:
	this.state.query === '' && this.inputField.blur();
	return this.clearQuery();
```

take it or leave it - just thought I'd share this fun-fact of how JavaScript short-circuiting logic works and how it can let us build conditional evaluation into a single line